src/main/java/com/selfxdsd/todocli/TodoVisitor.java

```diff
@@ -89,8 +97,14 @@ public FileVisitResult postVisitDirectory(
             final Path dir, final IOException exc
     ) throws IOException {
         if (dir.equals(root)) {
-            // scanning root has finished.
-            this.serializer.serialize();
+            try {
+                // scanning root has finished.
+                service.shutdown();
+                service.awaitTermination(5, TimeUnit.MINUTES);
```

Comment:
@criske Why do we wait for 5 minutes here?

Reply:
That's the timeout period, basically it says the service should finish its running and queued jobs within 5 minutes. Most likely we'll never reach that timeout.